### ISSUE TITLE
Attempt to invoke virtual method crash when exporting

### ISSUE THREAD

**Issue by voyteke** [SufficientlySecure/calendar-import-export#97]
When trying to export my calendar with around 1842 events, around half way through export applications crashes with

Error:
Attempt to  invoke virtual method ‘long java util get Time()'
on a null object‘ reference

I guess I have some 'bad' record?
How to resolve this?

That's on Chuwi tablet, just tried same calendar on Moto phone:

Same calendar shows 1765 records, not sure why different record count, also crashes with

Attempt to invoke virtual method ‘long net.fortuna,ical4.model.date.getTime()'
on a null object‘ reference

**Comment by fhaftmann** [comment]
This is very likely a deficiency in the export code.

Can you provide me with a minimal example? Usually it can be obtained by bisecting the calendar, i. e. dividing it in two roughly half parts, see which of those exposes the problem, and the continuing.

**Comment by voyteke** [comment]
Sure, ahem, how do I divide it, sorry, not sure how to do this..?
53% and 939/1788 it's more or less last values I think I see before the crash

**Comment by voyteke** [comment]
sorry, I should've figured that out, but, didn't try before...
OK, I've split at around 30,000/20,000 lines, copied both files to mobile, 
I then Load the file, is that it ? I get 1640 entries in first half, I thought I should be able to Export that file.. ?
I get option to Insert or Delete ?
sorry, not sure ... or am I meant to send/upload that to you ?
or do I make a new calendar, import into that, then, export ?

**Comment by fhaftmann** [comment]
The goal is to find out which entry makes problems.

Of course you would be free to send me the entire calendar file, but this would expose a lot of private data (possibly not only yours). Hence my proposal to aim for a minimal problematic calendar file, which in turn can be easily stripped of any personal data.

Following your description, it seems that the first half does not provoke any error. What about the second file? In my expectation it should, which would mean that the critical entry was part of it. After that identification, you can continue with the splitting until a calendar file with a single entry is reached, which should be the case after approx. 11 steps.

**Comment by fhaftmann** [comment]
During the course of writing, I had a thought that maybe I did not understand your problem right.

You are talking about _exporting_ an existing calendar on your mobile, not _importing_ an ical file into an calendar?

If that is the case, we have to devise a different approach.

But first lets make sure we have the same understaning here.

**Comment by voyteke** [comment]
yes, I'm attempting to EXPORT my calendar, so I may 'process' work records from my calendar in a s/s or the like

**Comment by fhaftmann** [comment]
OK, then my technical description was definitely misleading.

What is the technical representation of your calendar? A remote calendar hosted by a service provider?

Then we are faced with the difficulty how to get to a reproducible example then.

Does your service provider allow to duplicate your calendar? Then this could be used to get iteratively to a minimal example.

Once we have a minimal example, tinkering with the different options of the event could exhibit where the problem is.

(Sorry if this sounds very vague, but I can only extrapolate what is going on as long as I have no reproducible example to get hands on.)

**Comment by voyteke** [comment]
I'm using just standard consumer grade free Google calendar, with entries keyed in or copy/pasted from emails or web scrapes, is this what you're asking?
I noticed when I web scrape it often ends including html stuff, I'm guessing something for saved? that's causes that's error? I've came across a website that let's me export to CSV, which let's me do what I need, though i thought if I can do my export on my own, it would be good. I guess there is no way to export just a sub part of entries?

**Comment by fhaftmann** [comment]
The error is very likely caused by a field in an event containing unexpected (not necessarily »wrong«) data.

There is no way to export a subset of the entries, unfortunately.

Does Google offer the option to duplicate a calendar? This would open space for analysis.

**Comment by mcellist** [comment]
Hi there,

maybe I can catch this topic up, as I experience the very same problem unfortunately.

I have a google calendar with 4666 events.

The error message is exactly the same. Just by reading the message "long java util get Time()'
on a null object‘ reference":

_could there be a problem with an event having no time (hours) or an invalid time value?_

The point is, how to find the error causing event, as @fhaftmann already stated. Having a better idea what to look for, could help to find the event(s) in an export file crated by google (ical, I guess).

Can someone assist?

**Comment by woheller69** [comment]
I think a logcat would be needed to find the cause

**Comment by mcellist** [comment]
> I think a logcat would be needed to find the cause

How can I access it? Do I need a rooted device? I use a Samsung Android 12 device (no root).

**Comment by woheller69** [comment]
you need a PC and Android Debug Bridge.

https://developer.android.com/tools/adb

then

```adb logcat -D > log.txt``` and upload log.txt
